### PR TITLE
docs(architecture): document the uPortal Gradle task graph

### PR DESCRIPTION
**Pull request by ChristianMurphy**
##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]
-   [x] documentation is changed or added

##### Description of change

Include a copy of the uPortal release task graph.
Offer some insight into how to interpret the graphic.

:notebook: The task graph was captured during the uPortal 5.0.0 RC-1 release (thanks @drewwills) and was processed using https://github.com/Jasig/uPortal/pull/1026

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

**Comment by ChristianMurphy** [comment]
:eye: Preview https://github.com/ChristianMurphy/uPortal/blob/f6d7c7385a03076d511353b4062cce5908abea6c/docs/developer/other/ARCHITECTURE.md

File: docs/developer/other/ARCHITECTURE.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+:notebook: N.B. JavaDoc tasks do tend to be highly linked together, however cross-linking across JavaDoc does not necessarily signal tight coupling.
+
+### Tightly coupled Subprojects
+
+Projects with tight coupling tend to be pulled toward the center of the graph, and have many lines going outward from their `compileJava` task.
+For example: the `uPortal-security/uPortal-security-authn` subproject depends on 23 other subprojects.
+
+![tight coupling in task graph](../../images/release-task-graph-tight-coupling.png)
+
+:notebook: N.B. JavaDoc tasks do tend to be highly linked together, however cross-linking across JavaDoc does not necessarily signal tight coupling.
```
Comment (bjagg) [inline docs/developer/other/ARCHITECTURE.md]: Should this be stated once before the sub-project sections?

Comment (ChristianMurphy) [reply docs/developer/other/ARCHITECTURE.md]: Good idea, updated in 63f109f

**Comment by ChristianMurphy** [comment]
:eye: Direct link to SVG preview https://rawgit.com/ChristianMurphy/uPortal/63f109f73f39ecb81d778bbd97385796865b5270/docs/images/uportal-release-task.min.svg

**Pull request merged**